src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecBasedSpecProvider.cs

```diff
@@ -222,19 +225,55 @@ public void UpdateMergeTransitionInfo(long? blockNumber, UInt256? terminalTotalD
 
         public IReleaseSpec GenesisSpec => _transitions.Length == 0 ? null : _transitions[0].Release;
 
-        public IReleaseSpec GetSpec(ForkActivation forkActivation) =>
-                _transitions.TryGetSearchedItem(forkActivation,
+        public IReleaseSpec GetSpec(ForkActivation activation)
+        {
+            if (activation.Timestamp is null)
+            {
+                return _transitions.TryGetSearchedItem(activation,
                     CompareTransitionOnBlock,
                     out (ForkActivation, ReleaseSpec Release) transition)
                     ? transition.Release
-                    : null;
+                    : GenesisSpec;
+            }
+            List<(ForkActivation ForkActivation, IReleaseSpec Release)> filterdByBlockNumber = new();
+            long biggestBlockNumber = 0;
+            for (int i = 0; i < _transitions.Length; i++)
+            {
+                (ForkActivation ForkActivation, IReleaseSpec Release) transition = _transitions[i];
+                long currentBlockNumber = transition.ForkActivation.BlockNumber;
+                if (currentBlockNumber <= activation.BlockNumber && currentBlockNumber > biggestBlockNumber)
```

Comment:
Could we save the biggestBlockNumber in the constructor instead of doing it in every access to releaseSpec?

Reply:
Not really, it is dependent on the blockNumber and timestamp you are getting the spec for.